### DISCUSSION TITLE
How can I use ssh pubkey auth after the removal of host switcher?

### DISCUSSION THREAD

**Discussion by levi-jcbs** [cockpit-project/cockpit#20901] [Q&A] [answered]
### Explain what happens

Hi, Developers,

first of all, I want to thank you for your great work at this project. It's really helping me with my work as a server admin.

I have password authentication disabled on SSH & Cockpit on all of my servers and am using the host switcher feature on my local cockpit instance to connect to all of my servers via ssh pubkey authentication. Now, as we're facing the removal of this feature I'm asking myself if I can keep using cockpit with password auth disabled and if yes, how?

I would very much appreciate if you could suggest me a better place to discuss this topic if I'm in the wrong room here.

Thank you,
Levi

### Version of Cockpit

322

### Where is the problem in Cockpit?

Unknown or not applicable

### Server operating system

Debian

### Server operating system version

sid

### What browsers are you using?

Firefox, Chrome

**Comment by martinpitt** [comment]
Hello @levi-jcbs ! As said in the [release announcement](shttps://cockpit-project.org/blog/cockpit-322.html), the primary alternative for this is the [Cockpit Client flatpak](https://flathub.org/apps/details/org.cockpit_project.CockpitClient).

**Reply by levi-jcbs** [reply, marked as answer]
Thanks for your answer. The Cockpit Client Flatpak doesn't start on my debian sid machine, but I just learned about [TOTP PAM integration](https://github.com/google/google-authenticator-libpam).

I will access all my servers over their own domains and cockpit-webservers. To secure them I add the TOTP PAM module to the cockpit PAM config.

As you already said in the release announcement, this is probably the most secure way to use cockpit.
